Example/Tests/CollectionTestSortingTests.swift

```diff
@@ -0,0 +1,113 @@
+                return true
+            }
+            return false
+        }) { return $0.value < $1.value || ($0.value == $1.value && $0.rank > $1.rank) }
+        
+        //Sort by rank descending
+        let f2 = DiffCollectionFilter<CollectionTestObjectMock>(name: "Value starts with b", filter:{ s in
+            if(s.value.starts(with: "b")) {
+                return true
+            }
+            return false
+        }) { return $0.rank < $1.rank }
+        
+        tester = DiffCollectionTester(collection:DiffCollection(filters: [f1,f2]))
+    }
+    
+    override func tearDown() {
+        super.tearDown()
+    }
+    
+    //Add elements in random orders:
+    func test_Fillup_Collection() {
+        tester.testAdd(element: object4)
+        tester.testAdd(element: object7)
+        tester.testAdd(element: object9)
+        tester.testAdd(element: object1)
+        tester.testAdd(element: object12)
+        tester.testAdd(element: object6)
+        tester.testAdd(element: object2)
+        tester.testAdd(element: object8)
+        tester.testAdd(element: object3)
+        tester.testAdd(element: object10)
+        tester.testAdd(element: object11)
+        tester.testAdd(element: object5)
+    }
+    
+    func test_Filters_Providing_Sorting_Function_Must_Always_Remain_Sorted() {
+        test_Fillup_Collection()
+
+        tester.testElementAtIndexPathIsEqualTo(indexPath: IndexPath(row: 0, section: 0), element: object9)
+        tester.testElementAtIndexPathIsEqualTo(indexPath: IndexPath(row: 1, section: 0), element: object10)
+        tester.testElementAtIndexPathIsEqualTo(indexPath: IndexPath(row: 2, section: 0), element: object11)
+        tester.testElementAtIndexPathIsEqualTo(indexPath: IndexPath(row: 3, section: 0), element: object12)
+        tester.testElementAtIndexPathIsEqualTo(indexPath: IndexPath(row: 4, section: 0), element: object1)
+        tester.testElementAtIndexPathIsEqualTo(indexPath: IndexPath(row: 5, section: 0), element: object2)
+        tester.testElementAtIndexPathIsEqualTo(indexPath: IndexPath(row: 6, section: 0), element: object3)
+        tester.testElementAtIndexPathIsEqualTo(indexPath: IndexPath(row: 7, section: 0), element: object4)
+        
+        tester.testElementAtIndexPathIsEqualTo(indexPath: IndexPath(row: 0, section: 1), element: object8)
+        tester.testElementAtIndexPathIsEqualTo(indexPath: IndexPath(row: 1, section: 1), element: object7)
+        tester.testElementAtIndexPathIsEqualTo(indexPath: IndexPath(row: 2, section: 1), element: object6)
+        tester.testElementAtIndexPathIsEqualTo(indexPath: IndexPath(row: 3, section: 1), element: object5)
+    }
+    
+    func test_Adding_a_New_Object_Must_Inserted_at_Proper_Index() {
+        test_Fillup_Collection()
+        
+        var object = CollectionTestObjectMock(value: "zapato", status: .new, rank: 150)
+        tester.testAdd(element: object)
+        
+        //Must insert after element (zulu,430) at (4,0)
```

Comment:
How do we know that it should be at 4.0
The truth is we need to go and try to perform the code in our mind to be able to see that the index should be 4.0
But instead you should have created an expected result(array) and always compare with it. And also this creates a potential chance for a mistake. Make sense?

Reply:
I would use something like that(with separation logic into tester), where it's clear what to expect and no need to try to calculate the expected result. Easy to find the problem in case 
Notice that the tests should have less assertions. 

```swift
 func test_descending_order() {
        let filter = DiffCollectionFilter<Int>(name: "Simple Tests",
                                               filter: { $0 > 0 },
                                               sort: { $0 > $1 })
        var collection = DiffCollection(filters: [filter])
        collection.update(with: 1)
        collection.update(with: 2)
        collection.update(with: 5)
        collection.update(with: 7)
        collection.update(with: 4)
        
        XCTAssertEqual([1,2,4,5,7], collection.flatMap({ $0.map({$0} )}))
    }
```